eng/pipelines/templates/jobs/npm-release-task.yml

```diff
@@ -0,0 +1,133 @@
+            displayName: Get original tags
+          - task: EsrpRelease@9
+            displayName: 'Publish ${{parameters.Artifact.name}} to ESRP'
+            inputs:
+              ConnectedServiceName: 'Azure SDK PME Managed Identity'
+              ClientId: '5f81938c-2544-4f1f-9251-dd9de5b8a81b'
+              DomainTenantId: '975f013f-7f24-47e8-a7d3-abc4752bf346'
+              Usemanagedidentity: true
+              KeyVaultName: 'kv-azuresdk-codesign'
+              SignCertName: 'azure-sdk-esrp-release-certificate'
+              Intent: 'PackageDistribution'
+              ContentType: 'npm'
+              FolderLocation: ${{parameters.PathToArtifacts}}
+              Owners: ${{ coalesce(variables['Build.RequestedForEmail'], 'azuresdk@microsoft.com') }}
+              Approvers: ${{ coalesce(variables['Build.RequestedForEmail'], 'azuresdk@microsoft.com') }}
+              ServiceEndpointUrl: 'https://api.esrp.microsoft.com'
+              MainPublisher: 'ESRPRELPACMANTEST'
+              productstate: $(Tag)
+
+        - ${{ else }}:
+          - task: PowerShell@2
+            displayName: Publish to Dev Feed
+            inputs:
+              targetType: filePath
+              filePath: eng/tools/publish-to-npm.ps1
+              arguments: -pathToArtifacts ${{parameters.PathToArtifacts}} -accessLevel "public" -tag "${{parameters.Tag}}" -additionalTag "$(AdditionalTag)" -registry ${{parameters.Registry}} -npmToken $(azure-sdk-npm-token)
+              pwsh: true
+            condition: succeeded()
+
+- ${{ if eq(parameters.Registry, 'https://registry.npmjs.org/') }}:
+  - job: VerifyPackageTags
+    dependsOn:
+      - PublishPackage
+      - ${{ each dep in parameters.DependsOn }}:
+        - ${{ dep }}
+    condition: succeeded()
+    variables:
+      - template: /eng/pipelines/templates/variables/image.yml
+      - name: PackageName
+        value: $[dependencies.PublishPackage.outputs['PublishPackage.SetPackageVar.PackageName']]
+      - name: OriginalTags
+        value: $[dependencies.PublishPackage.outputs['PublishPackage.SetPackageVar.OriginalTags']]
+      - name: IntendedTagVersion
+        value: $[dependencies.PublishPackage.outputs['PublishPackage.SetPackageVar.IntendedTagVersion']]
+      - ${{ each var in parameters.VarDependencies }}:
+        - name: ${{ var.key }}
+          value: ${{ var.value }}
+    displayName: 'Verify package tags'
+    pool:
+      name: $(LINUXPOOL)
+      image: $(LINUXVMIMAGE)
+      os: linux
+    steps:
+    - checkout: self
+    - download: current
+    - task: Powershell@2
+      displayName: Verify package tags
+      inputs:
+        targetType: filePath
+        filePath: eng/scripts/verify-npm-tags.ps1
+        arguments: -originalDistTags '$(OriginalTags)' -intendedTag "$(Tag)" -intendedTagVersion "$(IntendedTagVersion)" -packageName "$(PackageName)" -npmToken "$(azure-sdk-npm-token)"
+        pwsh: true
+
+    - ${{ if ne(variables['AdditionalTag'], '') }}:
```

Comment:
I wouldn't expect this to work as the AdditionalTag variable isn't known at template compile time. I think you will need to make this a runtime condition instead.